app/grandchallenge/components/forms.py

```diff
@@ -106,6 +106,23 @@ class MultipleCIVForm(Form):
         SelectUploadWidget,
     }
 
+    @property
+    def cleaned_civ_data(self):
+        return {
+            k: v
+            for k, v in self.cleaned_data.items()
+            if k in self.base_obj.values_for_interfaces
+        }
+
+    @property
+    def cleaned_non_civ_data(self):
+        cleaned_civ_data = self.cleaned_civ_data
+        return {
+            k: v
+            for k, v in self.cleaned_data.items()
+            if k not in cleaned_civ_data
+        }
```

Comment:
I'm not sure, but I think this might not work. It could contain new CIVs (those that are dynamically added to the form) and those should go into `cleaned_civ_data`.

Reply:
Maybe doing it this way is too complex, and resorting to what you had initially, but then using a variable that you define in one place to define the keys to skip over, is best.

Reply:
Ooooooh, I hadn't thought of new interfaces =/! Yeah, there is no way to keep track of that in a simple matter!